### PR TITLE
zoom

### PR DESCRIPTION
**Pull request by ThisIsNSH**
Fixes issue #779

Changes: halfWave Schematic is made zoomable.

Screenshots for the change: 
ezgif com-video-to-gif 1
ezgif com-resize 1

**Note: The zoom +/- buttons can be removed**

APK for testing: 
[app-debug.zip](https://github.com/fossasia/pslab-android/files/1913298/app-debug.zip)

### CODE REVIEW

**Comment by ThisIsNSH** [comment]
@CloudyPadmal @akarshan96 @asitava1998 @viveksb007 please review.

**Comment by ThisIsNSH** [comment]
@mariobehling this time the a new class is created in ExperiementDocMdFragmentNew which handles markdowns and the image zoom.

File: app/src/main/java/org/fossasia/pslab/fragment/ExperimentDocMdFragmentNew.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ */
+
+public class ExperimentDocMdFragmentNew extends Fragment {
+
+    private NestedScrollView scrollView;
+    private String mdFile, mdFile1, imagePath;
+    private MarkdownView mMarkdownView, mMarkdownView1;
+    private WebView imageView;
+
+    public static org.fossasia.pslab.fragment.ExperimentDocMdFragmentNew newInstance(String mdFile, String mdFile1, String imagePath) {
```
[the remark this answers is not in the document]
Comment (ThisIsNSH) [reply app/src/main/java/org/fossasia/pslab/fragment/ExperimentDocMdFragmentNew.java]: Yes. I will refine the final code. How about the implementation? Does the zoom function correctly?
I couldn't understand your first comment.

File: app/src/main/java/org/fossasia/pslab/fragment/ExperimentDocMdFragmentNew.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import br.tiagohm.markdownview.MarkdownView;
+
+/**
+ * Created by ThisIsNSH on 4/15/2018.
+ */
+
+public class ExperimentDocMdFragmentNew extends Fragment {
+
+    private NestedScrollView scrollView;
+    private String mdFile, mdFile1, imagePath;
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/ExperimentDocMdFragmentNew.java]: @ThisIsNSH I meant in these cases. We can name them `mdFileTop`, `mdFileBottom` and so on!

Comment (ThisIsNSH) [reply app/src/main/java/org/fossasia/pslab/fragment/ExperimentDocMdFragmentNew.java]: 👍

**Comment by ThisIsNSH** [comment]
@CloudyPadmal please review. I have changed the variable names.

File: app/src/main/java/org/fossasia/pslab/fragment/ExperimentDocMdFragmentNew.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
+package org.fossasia.pslab.fragment;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.drawable.Drawable;
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/ExperimentDocMdFragmentNew.java]: Are these imports necessary? Check the Codacy status report and try to remove unused imports!

File: app/src/main/java/org/fossasia/pslab/fragment/ExperimentDocMdFragmentNew.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        mMarkdownTop.loadMarkdownFromAsset("DOC_HTML/apps/" + mdFileTop);
+        mMarkdownBottom.loadMarkdownFromAsset("DOC_HTML/apps/" + mdFileBottom);
+        webSvg.loadUrl("file:///android_asset/DOC_HTML/apps/images/schematics/" + svgPath);
+        webSvg.getSettings().setUseWideViewPort(false);
+        webSvg.getSettings().setBuiltInZoomControls(true);
+        webSvg.getSettings().setDisplayZoomControls(true);
+
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/ExperimentDocMdFragmentNew.java]: Avoid using empty extra lines

File: app/src/main/java/org/fossasia/pslab/fragment/ExperimentDocMdFragmentNew.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+    }
+
+    @Override
+    public void onDestroyView() {
+        super.onDestroyView();
+
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/ExperimentDocMdFragmentNew.java]: Empty lines

File: app/src/main/assets/DOC_HTML/apps/L_halfWavePart2.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+* CH1 monitors the original waveform output from W1.
+
+* CH2 monitors the rectified waveform .
+
+* Connect a load resistor to remove noise from the output.
+
+* To observe the effect of filtering, connect a capacitor to GND.
+
+* Connect various capacitors to study how their values affect the output signal.
+
```
Comment (CloudyPadmal) [inline app/src/main/assets/DOC_HTML/apps/L_halfWavePart2.md]: These extra lines can be avoided!

**Comment by ThisIsNSH** [comment]
@CloudyPadmal I made the suggested changes.

**Comment by ThisIsNSH** [comment]
[app-debug.zip](https://github.com/fossasia/pslab-android/files/1935267/app-debug.zip)
@CloudyPadmal I have improved the zoom functionality
* Due to WebView inside ScrollView the zoom was not proper.
* Now Jpg schematic is used inside TouchImageView to provide proper **Double tap zoom**

**Pull request closed without merging**